### PR TITLE
Update NetP settings item to match design

### PR DESCRIPTION
**Pull request by graeme**
Task/Issue URL: https://app.asana.com/0/0/1205084446087077/f
Tech Design URL:
CC:

**Description**:

For [iOS: Update NetworkProtection package to support iOS](https://app.asana.com/0/0/1204188117085830) we added a settings menu item to access NetP but it was done to a speculative design and with the incorrect fonts.

This task should:
Update the fonts to match the other settings items
Show Connected or Not connected depending on the PacketTunnelProvider`s status.

For the designs, check https://app.asana.com/0/0/1205084446087077/f

**Steps to test this PR**:
1. Check out this branch
2. Go to Build Phases and in Embed App Extensions, add the PacketTunnelProvider
3. Run on a real device
4. Ensure you’re authenticated as internal
5. Navigate to the Settings
6. **Check the Network Protection settings item is visible, says Not Connected and matches designs**
7. Tap the item to navigate to the NetP status view
8. If you have no token (there should be a message saying “Already Redeemed”), enter an[ invite code](https://app.asana.com/0/1203135672790568/1204303483337129/f)
9. Flip the switch to activate Network Protection
10. Navigate back to the settings screen
11. **Check the Network Protection settings item says Connected and matches designs**

<!—
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
—>

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `’`

**Orientation Testing**:

* [x] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [x] iOS 16

**Theme Testing**:

* [x] Light theme
* [ ] Dark theme

—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/Base.lproj/Settings.storyboard [hunk 1]
```diff
@@ -31,7 +31,7 @@
                                                 <label opaque="NO" multipleTouchEnabled="YES" contentMode="left" text="Set as Default Browser" textAlignment="natural" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" id="xof-5k-PkI">
                                                     <rect key="frame" x="20" y="0.0" width="334" height="43.666667938232422"/>
                                                     <autoresizingMask key="autoresizingMask"/>
-                                                    <fontDescription key="fontDescription" type="system" weight="regular" pointSize="16"/>
+                                                    <fontDescription key="fontDescription" type="system" pointSize="16"/>
```
Comment (diegoreymendez) [inline DuckDuckGo/Base.lproj/Settings.storyboard:34]: Are these changes intentional?

From the description of the PR I'd expect the NetP item would change to match the pre-existing fonts from other setting items.

File: DuckDuckGo/SettingsViewController.swift [hunk 2]
```diff
@@ -92,7 +96,10 @@ class SettingsViewController: UITableViewController {
     fileprivate lazy var featureFlagger = AppDependencyProvider.shared.featureFlagger
     fileprivate let syncService: DDGSyncing
     fileprivate let internalUserDecider: InternalUserDecider
-    private var internalStateChangeCancellable: Cancellable?
+#if NETWORK_PROTECTION
+    private let connectionObserver = ConnectionStatusObserverThroughSession()
```
Comment (diegoreymendez) [inline DuckDuckGo/SettingsViewController.swift:100]: I'm glad to know these things helped reusability. :)

Please take the following feedback as a thought on possible future direction.  There's no need to do anything about it as the current implementation is fine.

I think we should consider eventually abstracting this and https://github.com/duckduckgo/iOS/pull/1861/files#diff-fde6768638fd5f87f67d6aa963d581eac0ccf5ffcb9abe259e9dd5143a2e65c8R378-R392 a bit more into a view model.

**Review by diegoreymendez — COMMENTED** [review]
Added some comments.  I'm not approving yet because I'd like to understand the changes to the .storyboard file, since right now they seems to go a bit beyond the PR description.

File: DuckDuckGo/Base.lproj/Settings.storyboard (hunk 1 above)
Comment (graeme) [reply DuckDuckGo/Base.lproj/Settings.storyboard:34]: I don’t understand it either to be honest. I reverted it a couple of times and IB still seemed to change every item when I just tried to change the NetP one. I wonder if it’s because regular is now the default so there’s not need for the weight to be defined and IB now optimised the XML to reflect this… not sure.

Comment (diegoreymendez) [reply DuckDuckGo/Base.lproj/Settings.storyboard:34]: @graeme - Wow you're right, just saving the storyboard from develop with no changes did this for me too.  It seems something changed in the default storyboard specs that these changes are the new defaults.

**Review by diegoreymendez — APPROVED** [review]
Let's get this merged, nice work!
